lib/Common/Memory/Recycler.cpp

```diff
@@ -1298,6 +1314,28 @@ bool Recycler::AllowNativeCodeBumpAllocation()
 }
 
 
+#if DBG
+void Recycler::VerifyPageHeapFillAfterAlloc(char* memBlock)
```

Comment:
Consider passing in the attributes for the allocation here and asserting that either the found heap block is large or pageheap is not enabled for this object?